mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -218,6 +219,22 @@ class BooleanParam(parent: Params, name: String, doc: String) // No need for isV
   override def w(value: Boolean): ParamPair[Boolean] = super.w(value)
 }
 
+/** Specialized version of [[Param[Array[T]]]] for Java. */
+class ArrayParam[T : ClassTag](
```

Comment:
Having `ClassTag` is not Java friendly. `Array[T]` will be translated into `Object` in Java to handle both primitive arrays and object arrays. How about adding `StringArrayParam` instead of `ArrayParam[T]` in this PR?